pkg/imageengine/buildah/from.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	// TODO we do not support from remote currently
+	// which is to make the policy pull-if-missing
+	pullPolicy := define.PullNever
+
+	store := engine.ImageStore()
+
+	commonOpts, err := parse.CommonBuildOptions(engine.Command)
+	if err != nil {
+		return "", err
+	}
+
+	isolation, err := defaultIsolationOption()
+	if err != nil {
+		return "", err
+	}
+
+	namespaceOptions, networkPolicy := defaultNamespaceOptions()
+
+	usernsOption, idmappingOptions, err := parse.IDMappingOptions(engine.Command, isolation)
+	if err != nil {
+		return "", errors.Wrapf(err, "error parsing ID mapping options")
+	}
+	namespaceOptions.AddOrReplace(usernsOption...)
+
+	// hardcode format here, user do not concern about this.
+	format, err := getFormat(define.OCI)
+	if err != nil {
+		return "", err
+	}
+
+	capabilities, err := defaultContainerConfig.Capabilities("", []string{}, []string{})
+	if err != nil {
+		return "", err
+	}
+
+	commonOpts.Ulimit = append(defaultContainerConfig.Containers.DefaultUlimits, commonOpts.Ulimit...)
+
+	options := buildah.BuilderOptions{
+		FromImage:             opts.Image,
+		Container:             "",
+		ContainerSuffix:       "",
+		PullPolicy:            pullPolicy,
+		SystemContext:         systemContext,
+		DefaultMountsFilePath: "",
+		Isolation:             isolation,
+		NamespaceOptions:      namespaceOptions,
+		ConfigureNetwork:      networkPolicy,
+		CNIPluginPath:         "",
+		CNIConfigDir:          "",
+		IDMappingOptions:      idmappingOptions,
+		Capabilities:          capabilities,
+		CommonBuildOpts:       commonOpts,
+		Format:                format,
+		DefaultEnv:            defaultContainerConfig.GetDefaultEnv(),
+		MaxPullRetries:        maxPullPushRetries,
+		PullRetryDelay:        pullPushRetryDelay,
+		OciDecryptConfig:      &encconfig.DecryptConfig{},
+	}
+
+	if !opts.Quiet {
+		options.ReportWriter = os.Stderr
+	}
+
+	builder, err := buildah.NewBuilder(getContext(), store, options)
+	if err != nil {
+		return "", err
+	}
+
+	if err := onBuild(builder, opts.Quiet); err != nil {
+		return "", err
+	}
+
+	return builder.ContainerID, builder.Save()
+}
+
+func (engine *Engine) CreateContainer(opts *options.FromOptions) (string, error) {
+	wrapper := &fromFlagsWrapper{
+		FromAndBudResults: &buildahcli.FromAndBudResults{},
+		UserNSResults:     &buildahcli.UserNSResults{},
+		NameSpaceResults:  &buildahcli.NameSpaceResults{},
+	}
+
+	flags := engine.Flags()
+	fromAndBudFlags, err := buildahcli.GetFromAndBudFlags(wrapper.FromAndBudResults, wrapper.UserNSResults, wrapper.NameSpaceResults)
+	if err != nil {
+		return "", err
+	}
+
+	flags.AddFlagSet(&fromAndBudFlags)
+
+	err = engine.migrateFlags2BuildahFrom(opts)
+	if err != nil {
+		return "", err
+	}
+
+	return engine.createContainerFromImage(opts)
+}
+
+func (engine *Engine) migrateFlags2BuildahFrom(opts *options.FromOptions) error {
+	return nil
+}
+
+func onBuild(builder *buildah.Builder, quiet bool) error {
```

Comment:
I suggest that we define this function like the following way:
```
func (builder *buildah.Builder) Build(quiet bool) error
```

Reply:
This function if copied from buildah.